### PR TITLE
feat(userconfig/desc): add entity and availability types

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

- adds entity and availability types to the description builder
- removes the `BetaDescription` helper that we used previously
- adjusts old resources/datasources for the new implementation

## Why this way

this could be useful for optimizing the resources and effort that we need to make the docs

### CODE REVIEW

File: internal/schemautil/userconfig/desc.go [hunk 1]
```diff
@@ -5,23 +5,72 @@ import (
...
+const (
+	// Resource is a constant that represents the resource entity type.
+	Resource EntityType = iota
+	// DataSource is a constant that represents the data source entity type.
+	DataSource
+)
+
+// String is a function that returns the string representation of the entity type.
+func (et EntityType) String() string {
+	return [...]string{"resource", "datasource"}[et]
```
Comment (staceysalamon-aiven) [inline internal/schemautil/userconfig/desc.go:20]:
```suggestion
	return [...]string{"resource", "data source"}[et]
```

**Review by staceysalamon-aiven — APPROVED** [review]
Thanks so much for this! Just a very minor change to add a space to "data source".

File: internal/schemautil/userconfig/desc.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/desc.go]: Done. Sorry about that, we call it `datasource` in the code :D

**Review by byashimov — APPROVED** [review]
lgtm

**Pull request merged**